src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComObject.cs

```diff
@@ -77,10 +81,11 @@ internal ComObject(IIUnknownInterfaceDetailsStrategy interfaceDetailsStrategy, I
         /// </remarks>
         public void FinalRelease()
         {
-            if (UniqueInstance)
+            if (UniqueInstance && Interlocked.CompareExchange(ref _released, 1, 0) == 0)
             {
                 CacheStrategy.Clear(IUnknownStrategy);
                 IUnknownStrategy.Release(_instancePointer);
+                GC.SuppressFinalize(this);
```

Comment:
If UniqueInstance is false, the finalizer always runs and does the same Clear/Release operation?

Reply:
Yes, if UniqueInstance is false, we'll clear the caches in the finalizer (which will only run when the object is also removed from any caches in the ComWrappers infrastructure).